pkg/debuginfo/metadata.go

```diff
@@ -27,36 +28,37 @@ import (
 )
 
 var (
-	ErrMetadataShouldExist            = errors.New("debug info metadata should exist")
-	ErrMetadataExpectedStateUploading = errors.New("debug info metadata state should be uploading")
-	ErrMetadataUnknownState           = errors.New("debug info metadata state is unknown")
+	ErrMetadataShouldExist     = errors.New("debug info metadata should exist")
+	ErrMetadataUnexpectedState = errors.New("debug info metadata state is unexpected")
+	// There's no debug info metadata. This could mean that an older version
+	// uploaded the debug info files, but there's no record of the metadata, yet.
+	ErrMetadataNotFound = errors.New("debug info metadata not found")
 )
 
 type metadataState int64
 
 const (
 	metadataStateUnknown metadataState = iota
-	// There's no debug info metadata. This could mean that an older version
-	// uploaded the debug info files, but there's no record of the metadata, yet.
-	metadataStateEmpty
```

Comment:
The reason why this state exists is to be able to backfill the metadata files in the future, but happy to delete it if it's not needed!

Reply:
The unknown state or lack of the metadata file implies this state. Since we don't have a lot of historical data, this shouldn't be a huge problem, don't you think?

Reply:
Thought it would be nice to have an explicit state for this situation, but it's not an issue to leave it like this :)